File: src/main/java/coza/trojanc/receipt/context/ContextVariable.java
package coza.trojanc.receipt.context;

/**
 * An interface defining the basic functions required for a <code>ContextVariable</code> implementation.
 * A context variable is a variable that will be resolved by using the <code>expression</code> and input variables.
 * @author Charl Thiem
 */
public interface ContextVariable {

	/**
	 * Gets type.
	 *
	 * @return the type
	 */
	DynamicType getType();

	/**
	 * Gets key.
	 *
	 * @return the key
	 */
	String getKey();

	/**
	 * Gets formatting.
	 *
	 * @return the formatting
	 */
	String getFormatting();

	/**
	 * Gets expression.
	 *
	 * @return the expression
	 */
	String getExpression();
}

File: receipt-shared/src/main/kotlin/coza/trojanc/receipt/context/ContextVariable.kt
package coza.trojanc.receipt.context

/**
 * An interface defining the basic functions required for a `ContextVariable` implementation.
 * A context variable is a variable that will be resolved by using the `expression` and input variables.
 * @author Charl Thiem
 */
interface ContextVariable {

    /**
     * Gets type.
     *
     * @return the type
     */
    fun getType(): DynamicType

    /**
     * Gets key.
     *
     * @return the key
     */
    fun getKey(): String

    /**
     * Gets formatting.
     *
     * @return the formatting
     */
    fun getFormatting(): String?

    /**
     * Gets expression.
     *
     * @return the expression
     */
    fun getExpression(): String
}
